Second deliverable

FernandoGal25 opened the pull request:
- Creates rest repository to request from external api
- Creates new create pokemon use case
- Updates csv repository to persist new records

Commit:
Starts contruction of POST endpoint
- Creates rest repository to request from external api
- Creates new create pokemon use case
- Updates Mcsv repository to persist new records

RolloCasanova reviewed (changes requested):
Your PR changes are intended to be merged into the main branch (or the first deliverable), but not into your same branch.

For this reason, I can only see your changes in your last commit, but not from the previous one, and making it impossible to add comments there.

Please fix it so I can give it a proper review to these changes.

FernandoGal25 commented:
Made a new PR, here is the link: https://github.com/FernandoGal25/academy-go-q42021/pull/4

Do I need to fill a new form?

Thanks.

RolloCasanova commented:
It's okay, no need to fill the form again.

Commit:
Corrections of write endpoint
- Changes name of files to avoid redundancy eg.: controller/pokemon_controller.go => controller/pokemon.go
- Wraps errors in multiple wrappers in order to track the source
- Adds case use unit testing
- CCorrects documentation of ecode

Commit:
Adds error wrappers in rest pokemon repository

Commit:
Adds default config.yml
